src/Symfony/Component/Messenger/Transport/Doctrine/Connection.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Transport\Doctrine;
+
+use Doctrine\DBAL\Connection as DriverConnection;
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\DBAL\Schema\Synchronizer\SingleDatabaseSynchronizer;
+use Doctrine\DBAL\Types\Type;
+
+/**
+ * @author Vincent Touzet <vincent.touzet@gmail.com>
+ */
+class Connection
+{
+    /** @var string */
+    private $tableName;
+    /** @var \Doctrine\DBAL\Driver\Connection */
+    private $driverConnection;
+    /** @var bool */
+    private $debug;
+
+    public function __construct(string $tableName, DriverConnection $driverConnection, $debug = false)
+    {
+        $this->tableName = $tableName;
+        $this->driverConnection = $driverConnection;
+        $this->debug = $debug;
+    }
+
+    public function publish($body, $headers)
+    {
+        if (!$this->debug) {
+            $this->setup();
+        }
+        $query = sprintf(
+            'INSERT INTO %s (`body`, `headers`) VALUES (%s, %s)',
```

Comment:
Looks kinda forceful to force how the table is set, but I guess it's because it's a "provided doctrine connection" or something along those lines

Reply:
The connection is retrieved from the registry (See `DoctrineTransportFactory`)